src/components/header.tsx

```diff
@@ -23,9 +22,21 @@ interface HeaderProps {
 export const Header: FunctionComponent<HeaderProps> = ({ isConnected, setIsConnected }) => {
   const [wallet, setWalletInfo] = useState({} as walletInfoType);
 
+  const getWalletDetails = useCallback(async () => {
+    const walletDetails = await getWalletData();
+    if (walletDetails) {
+      setWalletInfo({
+        walletAddress: walletDetails.address,
+        walletNetwork: walletDetails.network,
+        walletBalance: walletDetails.balance,
+      });
+      setIsConnected(true);
+    }
+  }, [setIsConnected]);
+
   useEffect(() => {
     if (window.ethereum) {
-      window.ethereum.on("accountsChanged", function (accounts: Array<string>) {
+      window.ethereum.on("accountsChanged ", function (accounts: Array<string>) {
```

Comment:
dont need to clean callbacks? I already made this feedback

Reply:
I retaining the event listener but removing the useEffects.

Reply:
No, you must clean event listeners otherwise it's consider a memory leak.

Looks like there is a removeListener
- https://eips.ethereum.org/EIPS/eip-1193
- https://ethereum.stackexchange.com/questions/88084/how-to-clean-up-unused-meta-mask-event-listeners